### PR TITLE
[h265d/UMC]Fix saving SPS/PPS

### PR DESCRIPTION
**Pull request by akokoshn**
-If RawHeader_H265::m_buffer is empty - call Resize() and copy current header
 to m_buffer

### CODE REVIEW

File: _studio/shared/umc/codec/h265_dec/src/umc_h265_mfx_supplier.cpp [hunk 1]
```diff
@@ -280,24 +280,20 @@ UMC::Status MFXTaskSupplier_H265::DecodeHeaders(UMC::MediaDataEx *nalUnit)
...
+                        bool changed =
+                            size + prefix_size != hdr->GetSize() ||
+                            !!memcmp(hdr->GetPointer() + prefix_size, nalUnit->GetDataPointer(), size);
+
+                        if (isSPS && currSPS != nullptr)
+                            currSPS->m_changed = changed;
+                        else if (currPPS != nullptr)
+                            currPPS->m_changed = changed;
                     }
+                    hdr->Resize(id, size + prefix_size);
```
Comment (onabiull) [inline _studio/shared/umc/codec/h265_dec/src/umc_h265_mfx_supplier.cpp]: Given code above hdr may be nullptr?

Comment (akokoshn) [reply _studio/shared/umc/codec/h265_dec/src/umc_h265_mfx_supplier.cpp]: No, hdr->GetPointer() == nullptr means that hdr->m_buffer is empty.
hdr->Resize() make resize this vector for actual header
copy - filling

**Comment by akokoshn** [comment]
@dmitryermilov, the problem  - we don't save current SPS/PPS to hdr, because without call hdr->Resize() hdr->GetPointer == 0.

File: _studio/shared/umc/codec/h265_dec/src/umc_h265_mfx_supplier.cpp [hunk 2]
```diff
@@ -280,24 +280,20 @@ UMC::Status MFXTaskSupplier_H265::DecodeHeaders(UMC::MediaDataEx *nalUnit)
...
-                            else if (currPPS != nullptr)
-                                currPPS->m_changed = changed;
-                        }
-
-                        hdr->Resize(id, size + prefix_size);
-                        std::copy(start_code_prefix, start_code_prefix + prefix_size, hdr->GetPointer());
-                        std::copy((uint8_t*)nalUnit->GetDataPointer(), (uint8_t*)nalUnit->GetDataPointer() + size, hdr->GetPointer() + prefix_size);
+                        bool changed =
+                            size + prefix_size != hdr->GetSize() ||
+                            !!memcmp(hdr->GetPointer() + prefix_size, nalUnit->GetDataPointer(), size);
```
Comment (lakulako) [inline _studio/shared/umc/codec/h265_dec/src/umc_h265_mfx_supplier.cpp]: what for "!!" ? memcmp result is implicitly converted to bool.

**Pull request merged**